repos/spack_repo/builtin/packages/charliecloud/package.py

```diff
@@ -32,6 +34,8 @@ class Charliecloud(AutotoolsPackage):
     variant("cdi", default=True, description="Build with CDI support", when="@0.40:")
 
     depends_on("c", type="build")  # generated
+    depends_on("bdw-gc", type=("build", "link"))
```

Comment:
Did fd91121 miss removing this line, since the following line is more specific?

Reply:
Hello @reidpr, These two lines together are saying "This package always depends on `bdw-gc`. It depends on `bdw-gc` version 8 or later when this package is version 0.40 or later." Removing this line would change this to "This package only depends on `bdw-gc` *at all* when this package is version 0.40 or later. In that case, it depends on `bdw-gc` version 8 or later."